compiler/src/iree/compiler/Dialect/Flow/Transforms/ConvertLinalgMatmulToMmt4D.cpp

```diff
@@ -331,45 +332,58 @@ LinalgMatmulOpToLinalgMmt4DOpPattern::chooseTileParams(Value lhs, Value rhs,
   int64_t shapeN = rhsType.getShape()[1];
   auto chooseMatMulOrMatVec = [=](ArrayRef<int> m0k0n0,
                                   ArrayRef<int> m0k0n0ForMatVec,
+                                  ArrayRef<int> m0k0n0ForWhenRhsHas2Columns,
                                   std::string comment) {
     assert(m0k0n0ForMatVec[2] == 1 && "not a matrix*vector shape");
-    if (shapeN == 1) {
-      return Mmt4DTileParams(m0k0n0ForMatVec, comment + ", matrix*vector");
-    } else if (shapeM == 1) {
+    assert(m0k0n0ForWhenRhsHas2Columns[2] == 2 && "not a matrix*vector2 shape");
```

Comment:
The reader will wonder what a `vector2` is. I would say "N=2 is expected when RHS has 2 columns". Though this is literally repeating the assert condition, which is already printed by `assert`. This is why in my own projects I never write assert messages and just let the assert condition itself be self-explanatory. Up to you.

Reply:
Sorry that I forgot to update the comment when doing the change.. Yes, it's a bit annoying since the check already describes what's happening.. I updated the error message for consistency. (we also have the messages in other assertions.)